cmd/podman/machine/set.go

```diff
@@ -53,5 +90,26 @@ func setMachine(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	return vm.Set(vmName, setOpts)
+	if cmd.Flags().Changed("rootful") {
+		setOpts.Rootful = &setFlags.Rootful
+	}
+	if cmd.Flags().Changed("cpus") {
+		setOpts.CPUs = &setFlags.CPUs
+	}
+	if cmd.Flags().Changed("memory") {
+		setOpts.Memory = &setFlags.Memory
+		fmt.Println("memory")
+	}
+	if cmd.Flags().Changed("disk-size") {
+		setOpts.DiskSize = &setFlags.DiskSize
+	}
+
+	setErrs, err := vm.Set(vmName, setOpts)
+
+	if len(setErrs) > 0 {
```

Comment:
No need for this check.

Reply:
I would check for err != nil and return it, and then go through the list of setErrs?  Or do they get set and return err != nil?

Reply:
setErrors are errors while setting the settings. There can be more than one error, or none at all, ie, if changing --cpus and --memory both error, then both errors will be in the setErrs list. 

The other err is if writing the machine config causes an error. This a little worse because it might mean settings are set but not updated in the machine config.